### PR TITLE
Rwc2019

### PR DESCRIPTION
**Pull request by PetervDooren**

### CODE REVIEW

File: src/kinect/ray_tracer.cpp [hunk 1]
```diff
@@ -45,7 +48,7 @@ RayTraceResult ray_trace(const ed::WorldModel& world, const geo::Pose3D& raytrac
 
   geo::LaserRangeFinder lrf;
   lrf.setAngleLimits(-1e-3, 1e-3);
-  lrf.setNumBeams(1);
+  lrf.setNumBeams(2); // Because of Matthijs geolib refactor
```
Comment (MatthijsBurgh) [inline src/kinect/ray_tracer.cpp]: This should be fixed by https://github.com/tue-robotics/geolib2/pull/30

File: src/kinect/ray_tracer.cpp [hunk 2]
```diff
@@ -58,8 +61,24 @@ RayTraceResult ray_trace(const ed::WorldModel& world, const geo::Pose3D& raytrac
...
+        opt.setMesh(shape->getMesh(), raytrace_pose.inverse() * e->pose());
+
+        res.active_entity_ = e->id().str();
+        lrf.render(opt, res);
+      }
+    }
+
     geo::LaserRangeFinder::RenderOptions opt;
-    opt.setMesh(e->shape()->getMesh(), raytrace_pose.inverse() * e->pose());
+//    opt.setMesh(e->shape()->getBoundingBox().getMesh(), raytrace_pose.inverse() * e->pose()); // Use bbx
```
Comment (MatthijsBurgh) [inline src/kinect/ray_tracer.cpp]: Remove commented code

File: src/kinect/ray_tracer.cpp [hunk 3]
```diff
@@ -58,8 +61,24 @@ RayTraceResult ray_trace(const ed::WorldModel& world, const geo::Pose3D& raytrac
...
+    {
+      std::string name = volume.first;
+      geo::ShapeConstPtr shape = volume.second;
+      if (name == "on_top_of")
+      {
+        ROS_INFO("Raytrace on_top_of array of %s included", e->id().c_str());
+        geo::LaserRangeFinder::RenderOptions opt;
+        opt.setMesh(shape->getMesh(), raytrace_pose.inverse() * e->pose());
+
+        res.active_entity_ = e->id().str();
```
Comment (MatthijsBurgh) [inline src/kinect/ray_tracer.cpp]: res could be set once.

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]

File: src/kinect/ray_tracer.cpp (hunk 1 above)
Comment (MatthijsBurgh) [reply src/kinect/ray_tracer.cpp]: Reverted as https://github.com/tue-robotics/geolib2/pull/30 is merged.

File: src/kinect/kinect_plugin.cpp [hunk 4]
```diff
@@ -204,56 +204,83 @@ bool KinectPlugin::srvUpdate(ed_sensor_integration::Update::Request& req, ed_sen
...
+
+    visualization_msgs::Marker marker_msg;
+    marker_msg.header.frame_id = "map";
+    marker_msg.header.stamp = ros::Time::now();
+    marker_msg.action = visualization_msgs::Marker::ADD;
+    marker_msg.color.a = 0.5;
+    marker_msg.lifetime = ros::Duration(10.0);
+    marker_msg.scale.x = 0.05;
+
+    static int iter = 0;
```
Comment (LoyVanBeek) [inline src/kinect/kinect_plugin.cpp:237]: Huh? Why this?

File: src/kinect/kinect_plugin.cpp [hunk 5]
```diff
@@ -204,56 +204,83 @@ bool KinectPlugin::srvUpdate(ed_sensor_integration::Update::Request& req, ed_sen
...
+        const ed::EntityConstPtr& e = *it;
+        if (e->hasFlag("highlighted"))
+            update_req_->removeFlag(e->id(), "highlighted");
+    }
+
+    //
+    ed::EntityConstPtr hightlighted_e = world_->getEntity(res.entity_id);
+    if (hightlighted_e)
+    {
+        // ToDo: RoboCup hack
```
Comment (LoyVanBeek) [inline src/kinect/kinect_plugin.cpp]: Can be removed right?

**Review by LoyVanBeek — DISMISSED** [review]
OK overall, but some cleanup needed

File: src/kinect/kinect_plugin.cpp (hunk 4 above)
Comment (MatthijsBurgh) [reply src/kinect/kinect_plugin.cpp:237]: indent fix only

**Review by MatthijsBurgh — APPROVED** [review]

File: src/kinect/kinect_plugin.cpp (hunk 5 above)
Comment (MatthijsBurgh) [reply src/kinect/kinect_plugin.cpp]: Removed

**Pull request merged**